megamek/src/megamek/client/ui/swing/minimap/Minimap.java

```diff
@@ -584,6 +597,26 @@ public void run() {
         }
     };
 
+    private final List<Line> movePathLines = new Vector<>();
+    public record Line(int x1, int y1, int x2, int y2, Color color, int round) {};
+    private final Color MOVE_PATH_COLOR = new Color(0, 0, 0, 128);
+
+    private void addMovePath(List<UnitLocation> unitLocations, Entity entity) {
+        // values equal or lower than 0 mean no persistence
+        if (GUIP.getMovePathPersistenceOnMiniMap() <= 0) {
+            return;
+        }
+        Coords previousCoords = entity.getPosition();
```

Comment:
getPosition can easily return null, e.g. when a unit is offboard (arty), a flyer goes offboard to return later, or a unit isnt deployed yet. A frequent cause of bugs. I don't know if this method can be called in that case, but I suggest giving it a test at least with an arty unit deployed offboard.

Reply:
A unit that fled might also lead to problems.